### PR TITLE
Refactor validate_query()

### LINKED ISSUE #156: QueryValidator in `das-serverless-function` does not accept a list of queries, therefore not allowing the use of a conjunction query (AND clause).

### PR DESCRIPTION
**Pull request by marcocapozzoli**
Resolves #156

### CODE REVIEW

**Review by andre-senna — APPROVED** [review]
Please wait for @levisingularity and @eddiebrissow approvals before merging

**Review by rafaellevidomuso — APPROVED** [review]

**Review by levisingularity — APPROVED** [review]

File: das-query-engine/validators/actions.py
Comment (Pedrobc89) [inline das-query-engine/validators/actions.py:80]: I believe this should be outside the for.
```suggestion
        return True
```

**Review by Pedrobc89 — CHANGES_REQUESTED** [review]

File: das-query-engine/tests/integration/handle/test_query_action.py
Comment (Pedrobc89) [inline das-query-engine/tests/integration/handle/test_query_action.py:40]: Success are good an all for testing, but testing the invalid paths are very important as well.
If so, you would've caught the `return True` inside the for.

File: das-query-engine/validators/actions.py
Comment (eddiebrissow) [inline das-query-engine/validators/actions.py:59]: This limits the query, such as `[{}, []] `or `[[], []]`. To ensure the parameters are correct, we can to perform a recursive check and run additional tests, or simply verify the top-level type as either a list or a dictionary.

Comment (andre-senna) [reply das-query-engine/validators/actions.py:59]: > simply verify the top-level type as either a list or a dictionary.

This sounds like a good plan for now. @angeloprobst , do you have any inputs here?

Comment (angeloprobst) [reply das-query-engine/validators/actions.py:59]: I think that a recursive validation like the one suggested by @Pedrobc89 in the [ticket](https://github.com/singnet/das-serverless-functions/issues/156) covers all scenarios.

**Pull request merged**
